index.js

```diff
@@ -162,22 +162,19 @@ class ExecutorQueue extends Executor {
             winston.info(`scheduler enqueuing job timestamp  >>  ${JSON.stringify(job)}`));
 
         this.multiWorker.start();
-        await this.scheduler.connect();
-        this.scheduler.start();
+        Promise.resolve().then(this.scheduler.connect()).then(this.scheduler.start());
 
         process.on('SIGTERM', () => {
-            try {
-                await this.multiWorker.end();
-            } catch (err) {
-                winston.error(`failed to end the worker: ${error}`);
-            }
-            try {
-                await this.scheduler.end();
-            } catch (err) {
+            Promise.resolve().then(() => {
+                this.multiWorker.end();
```

Comment:
this.multiWorker.end().then/catch should be better

Reply:
or consider calling Promise.all with both async functions

Reply:
I choose to use `this.multiWorker.end().then/catch` pattern.
Thank you.

Reply:
@catto 
I fixed! I choose to use `this.multiWorker.end().then/catch` pattern.
Thank you.